tests/cpp/test_segmentation.cpp

```diff
@@ -702,4 +704,29 @@ TEST_F(SegmentationTest, MultipleSegmentSetsInOneSegment) {
   EXPECT_THAT(runtime->fusionSegments()->groups(), SizeIs(2));
 }
 
+TEST_F(SegmentationTest, ForwardInputsToSegmenterSetIssue2658) {
+  // Disable mark aliases prepare pass, which might insert more segment_set
+  preseg_passes::OptimizationPassGuard<preseg_passes::MarkAliasesPreparePass>
+      optimization_guard(false);
+
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* vanilla_in = makeContigConcreteTensor({2, 3});
+  TensorView* in = relu(vanilla_in);
+  TensorView* seg_in = segment_set(in);
```

Comment:
In issue #2658 , we ended up forwarding the `relu` op into `segment_set`. Ended up having two pointwise kernel.